File: solutions/python/space-age/1/space_age.py
class SpaceAge:
    _EARTH_YEAR = 31_557_600
    
    def __init__(self, seconds):
        self.seconds = seconds

    def on_earth(self):
        return round(self.seconds / SpaceAge._EARTH_YEAR, 2)

    def on_mercury(self):
        return round(self.seconds / 0.2408467 / SpaceAge._EARTH_YEAR, 2)

    def on_venus(self):
        return round(self.seconds / 0.61519726 / SpaceAge._EARTH_YEAR, 2)

    def on_mars(self):
        return round(self.seconds / 1.8808158 / SpaceAge._EARTH_YEAR, 2)

    def on_jupiter(self):
        return round(self.seconds / 11.862615 / SpaceAge._EARTH_YEAR, 2)

    def on_saturn(self):
        return round(self.seconds / 29.447498 / SpaceAge._EARTH_YEAR, 2)

    def on_uranus(self):
        return round(self.seconds / 84.016846 / SpaceAge._EARTH_YEAR, 2)

    def on_neptune(self):
        return round(self.seconds / 164.79132 / SpaceAge._EARTH_YEAR, 2)

File: solutions/java/space-age/2/src/main/java/SpaceAge.java
class SpaceAge {
    private final double seconds, earthYear, merYear, venYear, marYear, jupYear, satYear, uraYear, nepYear;
    private static final double EARTH_YEAR_SECS = 365.25 * 24 * 60 * 60;
    private static final double MER_ORBIT_PERIOD = 0.2408467;
    private static final double VEN_ORBIT_PERIOD = 0.61519726;
    private static final double MAR_ORBIT_PERIOD = 1.8808158;
    private static final double JUP_ORBIT_PERIOD = 11.862615;
    private static final double SAT_ORBIT_PERIOD = 29.447498;
    private static final double URA_ORBIT_PERIOD = 84.016846;
    private static final double NEP_ORBIT_PERIOD = 164.79132;
    
    SpaceAge(double seconds) {
        this.seconds = seconds;
        earthYear = seconds / EARTH_YEAR_SECS;
        merYear = earthYear / MER_ORBIT_PERIOD;
        venYear = earthYear / VEN_ORBIT_PERIOD;
        marYear = earthYear / MAR_ORBIT_PERIOD;
        jupYear = earthYear / JUP_ORBIT_PERIOD;
        satYear = earthYear / SAT_ORBIT_PERIOD;
        uraYear = earthYear / URA_ORBIT_PERIOD;
        nepYear = earthYear / NEP_ORBIT_PERIOD;
    }

    double getSeconds() {
        return seconds;
    }

    double onEarth() {
        return earthYear;
    }

    double onMercury() {
        return merYear;
    }

    double onVenus() {
        return venYear;
    }

    double onMars() {
        return marYear;
    }

    double onJupiter() {
        return jupYear;
    }

    double onSaturn() {
        return satYear;
    }

    double onUranus() {
        return uraYear;
    }

    double onNeptune() {
        return nepYear;
    }

}
